packages/react/popover/src/Popover.tsx

```diff
@@ -295,32 +286,31 @@ const PopoverContentNonModal = React.forwardRef<PopoverContentTypeElement, Popov
 
 /* -----------------------------------------------------------------------------------------------*/
 
+type PopoverContentImplElement = React.ElementRef<typeof PopperPrimitive.Content>;
```

Comment:
Any reason why this one is higher up and not close to the props one?
Not sure if I missed others like this too earlier…

Reply:
I did it so all the `Props` types were together and the `Element` types were first. Otherwise the element type kinda gets lost:

```jsx
type FocusScopeProps = Radix.ComponentPropsWithoutRef<typeof FocusScope>;
type DismissableLayerProps = Radix.ComponentPropsWithoutRef<typeof DismissableLayer>;
type PopperContentProps = Radix.ComponentPropsWithoutRef<typeof PopperPrimitive.Content>;
type PopoverContentImplElement = React.ElementRef<typeof PopperPrimitive.Content>;
interface PopoverContentImplProps
  extends PopperContentProps,
    Omit<DismissableLayerProps, 'onDismiss'> {
	// ...
}
```

What would we prefer?

Reply:
I wouldn't mind this even:

```tsx
type FocusScopeProps = Radix.ComponentPropsWithoutRef<typeof FocusScope>;
type DismissableLayerProps = Radix.ComponentPropsWithoutRef<typeof DismissableLayer>;

type PopperContentProps = Radix.ComponentPropsWithoutRef<typeof PopperPrimitive.Content>;
type PopoverContentImplElement = React.ElementRef<typeof PopperPrimitive.Content>;
interface PopoverContentImplProps
  extends PopperContentProps,
    Omit<DismissableLayerProps, 'onDismiss'> {
	// ...
}
```

Reply:
I've done this everywhere else:

```jsx
type PopoverContentImplElement = React.ElementRef<typeof PopperPrimitive.Content>;
type PopperContentProps = Radix.ComponentPropsWithoutRef<typeof PopperPrimitive.Content>;
interface PopoverContentImplProps extends PopperContentProps {
	// ...
}
```

As in, the props that it extends are after the element type... and since `FocusScopeProps` and `DismissableLayerProps` are also props it extends, I put those in the middle too.

I'm happy to move all props that it extends above the element type but would need to change that in every file too for consistency. e.g:

```jsx
type FocusScopeProps = Radix.ComponentPropsWithoutRef<typeof FocusScope>;
type DismissableLayerProps = Radix.ComponentPropsWithoutRef<typeof DismissableLayer>;
type PopperContentProps = Radix.ComponentPropsWithoutRef<typeof PopperPrimitive.Content>;

type PopoverContentImplElement = React.ElementRef<typeof PopperPrimitive.Content>;
interface PopoverContentImplProps
  extends PopperContentProps,
    Omit<DismissableLayerProps, 'onDismiss'> {
	// ...
}
```

Reply:
I'm not fussed, let's leave it as is then as it's consistent already.

Reply:
One thing that I found somewhat annoying in some of the other files is that multiple components might re-use the same type like:

```jsx
type MenuLabelElement = React.ElementRef<typeof Primitive.div>;
type PrimitiveDivProps = Radix.ComponentPropsWithoutRef<typeof Primitive.div>;
type MenuLabelProps extends PrimitiveDivProps {};

// ... later i have to extend `PrimitiveDivProps` but it's declared randomly elsewhere in the file

type MenuSeparatorElement = React.ElementRef<typeof Primitive.div>;
type MenuSeparatorProps extends PrimitiveDivProps {};
```

This made me wonder if we should just declare all external types upfront at the top of the file like:

```jsx
type PrimitiveDivProps = Radix.ComponentPropsWithoutRef<typeof Primitive.div>;

/* -------------------------------------------------------------------------------------------
 * MenuLabel
 * ------------------------------------------------------------------------------------------- */
type MenuLabelElement = React.ElementRef<typeof Primitive.div>;
type MenuLabelProps extends PrimitiveDivProps {};

/* -------------------------------------------------------------------------------------------
 * MenuSeparator
 * ------------------------------------------------------------------------------------------- */
type MenuSeparatorElement = React.ElementRef<typeof Primitive.div>;
type MenuSeparatorProps extends PrimitiveDivProps {};
```

That way, this example you're referring to would be like:

```jsx
type FocusScopeProps = Radix.ComponentPropsWithoutRef<typeof FocusScope>;
type DismissableLayerProps = Radix.ComponentPropsWithoutRef<typeof DismissableLayer>;
type PopperContentProps = Radix.ComponentPropsWithoutRef<typeof PopperPrimitive.Content>;

// lots of other components...

/* -------------------------------------------------------------------------------------------
 * PopoverContentImpl
 * ------------------------------------------------------------------------------------------- */
type PopoverContentImplElement = React.ElementRef<typeof PopperPrimitive.Content>;
interface PopoverContentImplProps extends PopperContentProps, Omit<DismissableLayerProps, 'onDismiss'> {
	// ...
}
```

Thoughts?